src/font/bitmapfont.js

```diff
@@ -5,69 +5,73 @@
  *
  * Font / Bitmap font
  *
- * ASCII Table
- * http://www.asciitable.com/
- * [ !"#$%&'()*+,-./0123456789:;<=>?@ABCDEFGHIJKLMNOPQRSTUVWXYZ[\]^_`abcdefghijklmnopqrstuvwxyz]
- *
  * -> first char " " 32d (0x20);
  */
 (function () {
     /**
-     * a bitpmap font object
+     * a bitmap font object
      * @class
-     * @extends me.Font
+     * @extends me.Renderable
      * @memberOf me
      * @constructor
      * @param {String} font font name
-     * @param {Number|Object} size either a number value, or an object like { x : 16, y : 16 }
      * @param {Number} [scale=1.0]
-     * @param {Number} [firstChar=0x20] charcode for the first character in the font sheet. Default is the space character.
+     * @param {String} [textAlign=left]
+     * @param {String} [textBaseline=top]
+     * @example
+     * // Use me.loader.preload or me.loader.load to load assets
+     * me.loader.preload([
+     * { name: "arial", type: "binary" src: "data/font/arial.fnt" },
+     * { name: "arial", type: "image" src: "data/font/arial.png" },
+     * ])
+     * // Then create an instance of your bitmap font:
+     * var myFont = new me.BitmapFont("arial");
+     * // And draw it inside your Renderable, just like me.Font
+     * myFont.draw(renderer, "Hello!", 0, 0);
      */
-    me.BitmapFont = me.Font.extend(
+    me.BitmapFont = me.Renderable.extend(
     /** @scope me.BitmapFont.prototype */ {
         /** @ignore */
-        init : function (font, size, scale, firstChar) {
+        init : function (fontName, scale, textAlign, textBaseline) {
             /** @ignore */
             // scaled font size;
-            this.sSize = new me.Vector2d();
+            this.sSize = me.pool.pull("me.Vector2d", 0, 0);
+
+            var fontData = me.loader.getBinary(fontName);
+            this.fontImage = me.loader.getImage(fontName);
+
+            /**
+             * The instance of me.BitmapFontData
+             * @type {me.BitmapFontData}
+             * @name bitmapFontData
+             * @memberOf me.BitmapFont
+             */
+            this.bitmapFontData = new me.BitmapFontData();
+            this.bitmapFontData.parse(fontData);
```

Comment:
I'm not sure what you mean on this one. Are you thinking something like this in terms of a public facing API?

    var data = new me.BitmapFontData(me.loader.getBinary(fontName));
    data.parse(); // or whenever
    var font = new me.BitmapFont(data, me.loader.getImage(fontName));

Or do you mean something else?

Personally I don't think we should convert to JSON. For the tile loader it makes sense, as the editor supports JSON & XML, we're just loading one as a dataset instead of maintaining two feature equal parsers.